controllers/permissionController.js

```diff
@@ -0,0 +1,147 @@
+import models from "../models";
+import constants from "../helpers/constants";
+import { isString, isEmpty } from "../helpers/funcValidators";
+
+const {
+  OK,
+  INTERNAL_SERVER_ERROR,
+  CONFLICT,
+  BAD_REQUEST,
+  CREATED
+} = constants.statusCode;
+const { Permission } = models;
+class PermissionController {
+  static async grantPermission(req, res) {
+    try {
+      const {
+        resource,
+        createPermission,
+        readPermission,
+        updatePermission,
+        deletePermission
+      } = req.body;
+      const { roleId } = req.params;
+      if (isEmpty(resource)) {
+        return res.status(BAD_REQUEST).json({
```

Comment:
My last suggestion is to separate concert in your methods, validations should be done with a middleware!
The create method should only create elements but not validate...

Reply:
Done, Thanks